### DISCUSSION TITLE
when the same data is opened by two people: updating and deleting

### DISCUSSION THREAD

**Discussion by dharen008** [livewire/livewire#6307] [1. Help]
[https://app.screencast.com/9z15QYraFonQX](https://app.screencast.com/9z15QYraFonQX)

here's the video above,

The scenario here is when there is a two user opens the same data, first person deleted the that data and second person didnt know that the first person deleted the data and still editing it.

i dont know if this is a bug or its only my code

but whenever the data id is deleted the livewire will not read the function anymore even if i will add a dd() at the very first of the function

if you can see it i added a dd('here'); but it did not show up on this scenario thats why my $checkAllEmployee did not work because the it did not continue to read the function.

i already tried using every single Lifecycle hooks.

let me know if there is a problem on my code

here's my code
`		public function mount() {
		$this->employeeId     = request()->input( 'id' );
		$this->activeUser     = auth()->user();
		$this->activeEmployee = EmployeeModel::find( $this->activeUser->employee_id );

		if ( $this->employeeId ) {
			$this->thisData = EmployeeModel::find( $this->employeeId );

			// Real time check first
			$checkAllEmployee = EmployeeModel::where( 'id', $this->employeeId )->first();

			if ( !$checkAllEmployee ) {
				session()->flash( 'errorList', 'Employee not found' );
				return redirect( '/employee/register2' );
			}

			$this->fill( $this->thisData->toArray() );
			$this->dataStatus = 'Update';
		} else {
			$this->dataStatus = 'Create';
		}
	}
	
	public function update() {
		dd('HERE');
		// Real time check first
		$checkAllEmployee = $this->thisData::where( 'id', $this->employeeId )->first();

		if ( !$checkAllEmployee ) {
			session()->flash( 'errorList', 'Employee not found' );
			return redirect( '/employee/register2' );
		} else {
			$thisEmployee  = EmployeeModel::find( $this->employeeId );
			$thisUser      = User::where( 'employee_id', $this->employeeId )->first();
			$existingEmail = EmployeeModel::where( 'work_email', $this->work_email )
							->where( 'id', '!=', $this->employeeId )
							->first();
	
			if ( $existingEmail ) {
				// Duplicate email found
				return back()->with( 'errorList', 'Duplicate email found: ' . $this->work_email );
			}

			$thisEmployee->fill( $this->all() );
			$thisEmployee->last_updated_by = $this->activeEmployee->id;
			$thisEmployee->save();
		
			// Save User
			$thisUser->employee_id   = $thisEmployee->id;
			$thisUser->email_address = $thisEmployee->work_email;
			$thisUser->password      = Hash::make( $thisEmployee->password );
			$thisUser->save();

			session()->flash( 'success', 'Employee successfully updated' );
		}
	}`

**Comment by mcpuishor** [comment]
I think this is a matter of app architecture, has nothing to do with the framework. 

You could introduce a "lock for edit" if the effort to implement such feature is justified.

**Reply by dharen008** [reply]
Yes we also have that page access function, even though there is a lock for edit, there will be a little chance that this scenario might come, that's why the final validation will always be on PHP before processing the action. in Laravel controller I can simply use my if statement, but in livewire somehow how the function stop reading the codes
